applications/nrf_desktop/src/modules/dvfs.c

```diff
@@ -244,28 +253,36 @@ static bool app_event_handler(const struct app_event_header *aeh)
 
 		if (check_state(event, MODULE_ID(main), MODULE_STATE_READY)) {
 			__ASSERT_NO_MSG((dvfs_high_freq_bitmask & dvfs_medlow_freq_bitmask) == 0);
+			if (IS_ENABLED(CONFIG_DESKTOP_DVFS_STATE_INITIALIZING_ENABLE)) {
+				if (module_flags_check_zero(&req_modules_bm)) {
+					process_dvfs_states(DVFS_STATE_INITIALIZING, true);
+				}
+				get_req_modules(&req_modules_bm);
+			} else {
+				enum dvfs_frequency_setting required_freq =
+								check_required_frequency();
+				if (required_freq != current_freq) {
+					set_dvfs_freq(required_freq);
+				}
+			}
+
```

Comment:
this part should be moved after k_work initialization. Otherwise, you may cancel uninitialized work